dymos/utils/command_line.py

```diff
@@ -0,0 +1,127 @@
+import openmdao.utils.hooks as hooks
+import argparse
+import sys
+import os
+from dymos.run_problem import modify_problem, run_problem
+from unittest.mock import patch
+
+
+def _simple_exec(script_name, user_args):
+    """
+    Use this as executor for commands that run as Problem commands.
+
+    Parameters
+    ----------
+    script_name : string
+        Name of the script to run.
+    user_args: list of strings
+        Any user supplied arguments for the script.
+    """
+    # use patch to avoid writing to sys.argv and sys.path
+    with patch.object(sys, 'argv', [script_name] + user_args):
+        with patch.object(sys, 'path', [os.path.dirname(script_name)] + sys.path[1:]):
+
+            with open(script_name, 'rb') as fp:
+                code = compile(fp.read(), script_name, 'exec')
+
+            globals_dict = {
+                '__file__': script_name,
+                '__name__': '__main__',
+                '__package__': None,
+                '__cached__': None,
+            }
+
+            exec(code, globals_dict)
+
+    return globals_dict
+
+
+def dymos_cmd(argv=None):
+    # pre-parse sys.argv to split between before and after '--'
+    alt_args = argv if argv else sys.argv
+    if '--' in alt_args:
+        idx = alt_args.index('--')
+        sys_args = alt_args[:idx]
+        user_args = alt_args[idx + 1:]
+        argv = sys_args  # don't change sys.argv
+    else:
+        user_args = []
+
+    parser = argparse.ArgumentParser(description='Dymos Command Line Tool')
+    parser.add_argument('script', type=str,
+                        help='Python script that creates a Dymos problem to run')
+    parser.add_argument('-s', '--simulate', action='store_true',
+                        help='If given, perform simulation after solving the problem.')
+    parser.add_argument('-n', '--no_solve', action='store_true',
+                        help='If given, do not run the driver on the problem (simulate only)')
+    parser.add_argument('-t', '--solution', default=None,
```

Comment:
I'm fine with this for now but before we release this as a documented feature we might want to think of a different argument pair for loading a previous solution/restart data.